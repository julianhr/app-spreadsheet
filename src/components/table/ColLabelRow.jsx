import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'

import ColLabel from '../cells/ColLabel'


const Row = styled.div`
  display: flex;
  width: fit-content;
`

const InnerBorder = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  border-right: 1px solid ${props => props.theme.colors.cell.borderDark};
  border-bottom: 1px solid ${props => props.theme.colors.cell.borderDark};
  background: ${props => props.theme.colors.cell.labelBkg};
`

function ColLabelRow({ colLabels }) {
  const renderCornerCell = () => (
    <InnerBorder
      className='row-label-width col-label-height'
    />
  )

  const renderLabelCells = () => (
    new Array(colLabels.length).fill(0).map((_, i) => (
      <ColLabel
        key={i}
        label={colLabels[i]}
      />
    )
  ))

  return (
    <Row
      data-row='header'
    >
      {renderCornerCell()}
      {renderLabelCells()}
    </Row>
  )
}

ColLabelRow.propTypes = {
  colLabels: PropTypes.arrayOf(PropTypes.string)
}

export default ColLabelRow

