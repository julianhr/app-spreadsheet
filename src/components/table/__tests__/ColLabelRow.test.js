import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'

import MockApp from '~/__tests__/__mocks__/MockApp'
import ColLabelRow from '../ColLabelRow'


const renderApp = (props) => {
  const wrapper = render(
    <MockApp>
      <table>
        <ColLabelRow {...props} />
      </table>
    </MockApp>
  )

  return wrapper
}

describe('ColLabelRow', () => {
  afterEach(cleanup)

  it('matches snapshot', () => {
    const colLabels = ['A', 'B']
    const props = { colLabels }
    renderApp(props)
    const el = document.querySelector('[data-row="header"]')
    expect(el).toMatchSnapshot()
  })
})

