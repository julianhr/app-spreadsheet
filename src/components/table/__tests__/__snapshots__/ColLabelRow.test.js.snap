// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColLabelRow matches snapshot 1`] = `
<div
  class="css-g0yt5d-Row e1ia7370"
  data-row="header"
>
  <div
    class="row-label-width col-label-height css-ql6bmv-InnerBorder e1ia7371"
  />
  <div
    class="col-label-width col-label-height css-19eh4ji-InnerBorder e1al44mo0"
    data-col="A"
  >
    A
  </div>
  <div
    class="col-label-width col-label-height css-19eh4ji-InnerBorder e1al44mo0"
    data-col="B"
  >
    B
  </div>
</div>
`;
